nomos-services/blend/src/core/mod.rs

```diff
@@ -695,77 +730,132 @@ async fn run_event_loop<
     ProofsVerifier: ProofsVerifierTrait,
     RuntimeServiceId: Sync,
 {
-    // Main event loop that persists while the node is a core node.
-    // This can span across multiple sessions.
+    // An optional crypto processor to handle the old session during transition
+    // period.
+    let mut old_session_crypto_processor: Option<
+        CoreCryptographicProcessor<NodeId, ProofsGenerator, ProofsVerifier>,
+    > = None;
```

Comment:
I think it would be better to merge both crypto processors in a cryptor processor that handles sessions. It would be a wrapper of arbitrary cryptroprocessors and tries for each (?)